### ISSUE TITLE
Cannot use Annotator with PIL.Image

### ISSUE THREAD

**Issue by simone-gaiarin** [ultralytics/ultralytics#8364]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Other

### Bug

In `utils.plotting.Annotator`:

```
assert im.data.contiguous, "Image not contiguous. Apply np.ascontiguousarray(im) to Annotator() input images."
self.lw = line_width or max(round(sum(im.shape) / 2 * 0.003), 2)  # line width
```

These two lines assume the image is a cv2 image (numpy array) before checking the value of the `pil` parameter.
This makes it impossible to pass in `PIL.Image` objects.

### Environment

Ultralytics YOLOv8.1.7 🚀 Python-3.10.13 torch-2.1.2+cu121 CUDA:0 (NVIDIA RTX 2000 Ada Generation Laptop GPU, 7940MiB)
Setup complete ✅ (32 CPUs, 31.0 GB RAM, 702.6/934.3 GB disk)

### Minimal Reproducible Example

from PIL import Image
from ultralytics.utils.plotting import Annotator

image_path = 'path/to/your/image.jpg'
image = Image.open(image_path)

annotator = Annotator(image, pil=True)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@simone-gaiarin hi there! Thanks for bringing this to our attention. 🛠️

It looks like there's an assertion in the `Annotator` class that's expecting a NumPy array before it checks for the `pil` parameter. This indeed would prevent passing `PIL.Image` objects directly. A quick workaround would be to convert the PIL image to a NumPy array before passing it to the `Annotator`. Here's a code snippet for that:

```python
from PIL import Image
import numpy as np
from ultralytics.utils.plotting import Annotator

image_path = 'path/to/your/image.jpg'
image = Image.open(image_path)
image_np = np.array(image)

annotator = Annotator(image_np, pil=True)
```

We'll look into this issue and consider updating the `Annotator` class to handle `PIL.Image` objects more gracefully. If you have any further questions or suggestions, feel free to share! 👍

**Comment by glenn-jocher** [comment]
@simone-gaiarin great news 😃! Your original issue may now be resolved ✅ in PR #8397. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by simone-gaiarin** [comment]
I confirm it works with the proposed changes.
